vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -158,6 +203,68 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
         return dotnetPath;
     }
 
+    private async checkForPartialInstalls(version : string, uninstallLocalRuntime : boolean, uninstallLocalSDK : boolean)
+    {
+        const installingVersions = this.context.extensionState.get<string[]>(this.installingVersionsKey, []);
+        const partialInstall = installingVersions.indexOf(version) >= 0;
+        if (partialInstall)
+        {
+            // Partial install, we never updated our extension to no longer be 'installing'.
+            this.context.eventStream.post(new DotnetAcquisitionPartialInstallation(version));
+
+            // Uninstall everything so we can re-install. For global installs, let the installer handle it.
+            if(uninstallLocalRuntime)
+            {
+                await this.uninstallAll();
+            }
+            else if(uninstallLocalSDK)
+            {
+                await this.uninstallRuntime(version);
+            }
+        }
+    }
+
+    private async acquireGlobalCore(globalInstallerResolver : GlobalInstallerResolver): Promise<string>
+    {
+        const installingVersion = await globalInstallerResolver.getFullVersion();
+        this.checkForPartialInstalls(installingVersion, false, false);
+
+        const installer : IGlobalInstaller = os.platform() === 'linux' ?
+            new LinuxGlobalInstaller(this.context, installingVersion) :
+            new WinMacGlobalInstaller(this.context, installingVersion, await globalInstallerResolver.getInstallerUrl());
+
+        // Indicate that we're beginning to do the install.
+        await this.addVersionToExtensionState(this.installingVersionsKey, installingVersion);
+        this.context.eventStream.post(new DotnetAcquisitionStarted(installingVersion));
+
+        // See if we should return a fake path instead of running the install
+        if(process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH && process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH === 'true')
+        {
+            return 'fake-sdk';
+        }
+
+        const installerResult = await installer.installSDK();
+
+        if(installerResult !== '0')
+        {
+            const err = new DotnetNonZeroInstallerExitCodeError(new Error(`An error was raised by the .NET SDK installer. The exit code it gave us: ${installerResult}`));
```

Comment:
I wish there was something actionable we could tell the user to do here. E.G. check MSI logs for Windows, check /var/log/install.log for macOS, check /var/log/dpkg.log for Ubuntu, etc. I don't know of something immediately, but it might be nice if the different IGlobalInstaller implementations knew how to report where to look for logs/help, then we could consume that in this error.

Reply:
I agree. I think there should be a document detailing the meaning of certain exit codes that can from the dotnet installer. such as 1641 means pending reboot. We could link to that if it exists. Should we go and make an issue in dotnet docs? I dont know who can supply that information and I couldnt find it in the installer